### PR TITLE
[Update] 3rd party tools compatibility

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+**规避方法**
+
+使用额外的 `UPDATE` 语句进行数据更新，可使用事务保证数据一致性。
+
+### 不支持 `useLocalTransactionState` 参数
+
+**描述**
+
+TiDB 依赖 `COMMIT()` 与 `ROLLBACK()` 进行事务的提交或回滚。[useLocalTransactionState](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-performance-extensions.html) 参数可能会省略在 MySQL 场景下不必要的 `COMMIT()` 与 `ROLLBACK()`，但这对 TiDB 来说是必要的。
```
Comment (crazycs520) [inline develop/dev-guide-third-party-tools-compatibility.md:133]: “参数可能会省略在 MySQL 场景下不必要的 `COMMIT()` 与 `ROLLBACK()`，但这对 TiDB 来说是必要的。”

没太懂这个的意思

Comment (Icemap) [reply develop/dev-guide-third-party-tools-compatibility.md:133]: useLocalTransactionState 开启时在一定条件下会省略 `COMMIT()` 与 `ROLLBACK()`， 这个语句对 TiDB 来讲是必要的。这样说会不会好点？

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+`useLocalTransactionState` 和 `rewriteBatchedStatements` 两参数同时开启时，将导致事务无法提交的问题。你可以使用[代码](https://github.com/Icemap/tidb-java-gitpod/tree/reproduction-local-transaction-state-txn-error)复现。
+
+**规避方法**
+
+> **注意：**
+>
+> 已向 MySQL 报告此 Bug，可关注此 [Bug Report](https://bugs.mysql.com/bug.php?id=108643) 进行最新消息的跟踪。
```
Comment (winkyao) [inline develop/dev-guide-third-party-tools-compatibility.md:141]:
```suggestion
> 已向 MySQL JDBC 报告此 Bug，可关注此 [Bug Report](https://bugs.mysql.com/bug.php?id=108643) 进行最新消息的跟踪。
```

**Review by winkyao — APPROVED** [review]
LGTM

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+>
+> 这样的功能不兼容将被视为预期行为，将不再重复叙述。可查看[与 MySQL 兼容性对比](/mysql-compatibility.md)获得更多不支持或与 MySQL 不一致的特性说明。
+
+## 通用
+
+### `SELECT CONNECTION_ID()` 返回结果类型为 64 位整型
+
+**描述**
+
+TiDB 的 `SELECT CONNECTION_ID()` 返回值为 64 位，如 `2199023260887`。而 MySQL 为 32 位， 如 `391650`。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:28]:
```suggestion
TiDB 的 `SELECT CONNECTION_ID()` 返回值为 64 位，如 `2199023260887`。而 MySQL 为 32 位，如 `391650`。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+请注意使用各语言的 64 位整型类型(或字符串类型)进行接收，防止溢出。如 Java 应使用 `Long` 或 `String` 进行接收，JS/TS 应使用 `string` 类型进行接收。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:32]:
```suggestion
请注意使用各语言的 64 位整型类型（或字符串类型）存储 `SELECT CONNECTION_ID()` 的结果，防止溢出。如 Java 应使用 `Long` 或 `String` 进行接收，JavaScript/TypeScript 应使用 `string` 类型进行接收。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+### TiDB 未设置 `Com_*` 计数器
+
+**描述**
+
+MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 SELECT 语句数（即使语句并未成功执行）。而TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察两数据库的差异。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:38]:
```suggestion
MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察 TiDB 与 MySQL 的差异。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+请勿使用这样的变量，常见的使用场景如监控等。TiDB 的可观测性较为完善，无需从服务端变量进行查询。如需定制监控工具，可阅读 [TiDB 监控框架概述](/tidb-monitoring-framework.md)来获得更多信息。
+
+### TiDB 错误日志区分 `timestamp` 与 `datetime`，MySQL 不区分
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:44]:
```suggestion
### TiDB 错误日志区分 `TIMESTAMP` 与 `DATETIME` 类型，MySQL 不区分
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+TiDB 错误日志区分 `timestamp` 与 `datetime`，而 MySQL 不区分，全部返回为 `datetime`。即 MySQL 会将 `timestamp` 类型的报错信息错误地写为 `datetime` 类型。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:48]:
```suggestion
TiDB 错误日志区分 `TIMESTAMP ` 与 `DATETIME`，而 MySQL 不区分，全部返回为 `DATETIME`。即 MySQL 会将 `TIMESTAMP` 类型的报错信息错误地写为 `DATETIME` 类型。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+请勿使用错误日志进行字符串匹配，而是使用[错误码](/error-codes.md)进行故障诊断。
+
+### 不支持 `CHECK TABLE` 语句
+
+**描述**
+
+TiDB 使用 [ADMIN CHECK [TABLE|INDEX]](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。不支持 `CHECK TABLE` 语句。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:58]:
```suggestion
TiDB 使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。不支持 `CHECK TABLE` 语句。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+转为使用 `ADMIN CHECK [TABLE|INDEX]`。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:62]:
```suggestion
转为使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md)。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+## MySQL JDBC 不兼容
+
+- 测试版本：MySQL Connector/J 8.0.29
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:66]:
```suggestion
测试版本为 MySQL Connector/J 8.0.29
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+新版本的 MySQL Connector/J 在与 5.7.5 版本以下的 MySQL 服务端，或使用 5.7.5 版本以下 MySQL 服务端协议的数据库（如 TiDB 6.3 版本以下）一起工作时。将在一定情况下引起连接的挂起，可查看此 [Bug Report](https://bugs.mysql.com/bug.php?id=106252) 了解更细节的信息。
+
+**规避方法**
+
+这是一个已知的问题，MySQL Connector/J 至今未合并修复代码。
+
+PingCAP 对其进行了两个维度的修复：
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:155]:
```suggestion
TiDB 对其进行了两个维度的修复：
```

File: TOC.md [hunk 12]
```diff
@@ -78,6 +78,7 @@
     - [Gitpod](/develop/dev-guide-playground-gitpod.md)
   - 第三方软件支持
     - [TiDB 支持的第三方工具](/develop/dev-guide-third-party-support.md)
+    - [已知的第三方工具兼容问题](/develop/dev-guide-third-party-tools-compatibility.md)
```
[the remark this answers is not in the document]
Comment (Icemap) [reply TOC.md:81]: 这是为啥？之前不是需要全部的 `/develop/` 下的文件都增加 dev-guide- 的前缀嘛？

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
+---
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-third-party-tools-compatibility.md:1]: ditto

**Comment by Icemap** [comment]
[英文 PR](https://github.com/pingcap/docs/pull/10477) 已一并更新

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察 TiDB 与 MySQL 的差异。
+
+**规避方法**
+
+请勿使用这样的变量，常见的使用场景如监控等。TiDB 的可观测性较为完善，无需从服务端变量进行查询。如需定制监控工具，可阅读 [TiDB 监控框架概述](/tidb-monitoring-framework.md)来获得更多信息。
+
+### TiDB 错误日志区分 `TIMESTAMP` 与 `DATETIME` 类型，MySQL 不区分
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:44]:
```suggestion
### TiDB 错误日志区分 `TIMESTAMP` 与 `DATETIME` 类型
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+> **注意：**
+>
+> 已向 MySQL JDBC 报告此 Bug，可关注此 [Bug Report](https://bugs.mysql.com/bug.php?id=108643) 进行最新消息的跟踪。
+
+请勿开启 `useLocalTransactionState`，这有可能导致事务无法提交或回滚。
+
+### 新版本 Connector 无法兼容旧版本 Server
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:145]:
```suggestion
### Connector 无法兼容 5.7.5 版本以下的服务端
```
新、旧版本需要具体说明，是 Connector 8.0.29 或在标题中不写“新”

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+新版本的 MySQL Connector/J 在与 5.7.5 版本以下的 MySQL 服务端，或使用 5.7.5 版本以下 MySQL 服务端协议的数据库（如 TiDB 6.3.0 版本以下）一起工作时，将在一定情况下引起连接的挂起。关于更多细节信息，可查看此 [Bug Report](https://bugs.mysql.com/bug.php?id=106252)。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:149]:
```suggestion
MySQL Connector/J 8.0.29 在与 5.7.5 版本以下的 MySQL 服务端，或使用 5.7.5 版本以下 MySQL 服务端协议的数据库（如 TiDB 6.3.0 版本以下）同时使用时，将在一定情况下导致数据库连接的挂起。关于更多细节信息，可查看此 [Bug Report](https://bugs.mysql.com/bug.php?id=106252)。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+这是一个已知的问题，MySQL Connector/J 至今未合并修复代码。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:153]:
```suggestion
这是一个已知的问题，MySQL Connector/J 目前未合并修复代码。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+在 TiDB 应用程序中，请注意使用各语言的 64 位整型类型（或字符串类型）存储 `SELECT CONNECTION_ID()` 的结果，防止溢出。如 Java 应使用 `Long` 或 `String` 进行接收，JavaScript/TypeScript 应使用 `string` 类型进行接收。
+
+### TiDB 未设置 `Com_*` 计数器
+
+**描述**
+
+MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察 TiDB 与 MySQL 的差异。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:38]: SHOW GLOBAL STATUS LIKE 可以加个链接 https://docs.pingcap.com/zh/tidb/dev/sql-statement-show-status#show-globalsession-status

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+请勿使用这样的变量，常见的使用场景如监控等。TiDB 的可观测性较为完善，无需从服务端变量进行查询。如需定制监控工具，可阅读 [TiDB 监控框架概述](/tidb-monitoring-framework.md)来获得更多信息。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:42]: “常见的使用场景如监控等” 这句话和上句话不连贯。

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+
+### TiDB 错误日志区分 `TIMESTAMP` 与 `DATETIME` 类型，MySQL 不区分
+
+**描述**
+
+TiDB 错误日志区分 `TIMESTAMP` 与 `DATETIME`，而 MySQL 不区分，全部返回为 `DATETIME`。即 MySQL 会将 `TIMESTAMP` 类型的报错信息错误地写为 `DATETIME` 类型。
+
+**规避方法**
+
+请勿使用错误日志进行字符串匹配，而是使用[错误码](/error-codes.md)进行故障诊断。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:52]:
```suggestion
请勿使用错误日志进行字符串匹配，要使用[错误码](/error-codes.md)进行故障诊断。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+
+TiDB 使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。不支持 `CHECK TABLE` 语句。
+
+**规避方法**
+
+在 TiDB 中转为使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md)。
+
+## MySQL JDBC 不兼容
+
+测试版本为 MySQL Connector/J 8.0.29
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:66]:
```suggestion
测试版本为 MySQL Connector/J 8.0.29。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+不设置 `enablePacketDebug` 参数。
+
+### 不支持 UpdatableResultSet
+
+**描述**
+
+TiDB 暂不支持 UpdatableResultSet，即请勿指定 `ResultSet.CONCUR_UPDATABLE` 参数，并在 ResultSet 内部进行数据更新。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:123]: 并在 ResultSet 内部进行数据更新。这句话是不建议做、还是可以做的？如果不建议做，考虑改成下面这样：
```suggestion
TiDB 暂不支持 UpdatableResultSet，即请勿指定 `ResultSet.CONCUR_UPDATABLE` 参数，也不要在 ResultSet 内部进行数据更新。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+
+`useLocalTransactionState` 和 `rewriteBatchedStatements` 两参数同时开启时，将导致事务无法提交的问题。你可以使用[代码](https://github.com/Icemap/tidb-java-gitpod/tree/reproduction-local-transaction-state-txn-error)复现。
+
+**规避方法**
+
+> **注意：**
+>
+> 已向 MySQL JDBC 报告此 Bug，可关注此 [Bug Report](https://bugs.mysql.com/bug.php?id=108643) 进行最新消息的跟踪。
+
+请勿开启 `useLocalTransactionState`，这有可能导致事务无法提交或回滚。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:143]: 标题只是说导致无法提交，这是是否要把回滚删掉，保持一致？

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+
+### TiDB 不支持 `CHECK TABLE` 语句
+
+**描述**
+
+TiDB 使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。不支持 `CHECK TABLE` 语句。
+
+**规避方法**
+
+在 TiDB 中转为使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md)。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:62]:
```suggestion
TiDB 不支持 `CHECK TABLE` 语句。

**规避方法**

在 TiDB 中使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+
+测试版本为 MySQL Connector/J 8.0.29
+
+### 默认排序规则不一致
+
+**描述**
+
+MySQL Connector/J 的排序规则保存在客户端内，通过获取的服务端版本进行判别。
+
+已知的客户端与服务端排序规则不一致的字符集：
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:74]:
```suggestion
下表列出了已知的客户端与服务端排序规则不一致的字符集：
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+
+### TiDB 未设置 `Com_*` 计数器
+
+**描述**
+
+MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察 TiDB 与 MySQL 的差异。
+
+**规避方法**
+
+请勿使用这样的变量，常见的使用场景如监控等。TiDB 的可观测性较为完善，无需从服务端变量进行查询。如需定制监控工具，可阅读 [TiDB 监控框架概述](/tidb-monitoring-framework.md)来获得更多信息。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:42]:
```suggestion
请勿使用这样的变量。在 MySQL 中 `Com_*` 常见的使用场景之一是监控。TiDB 的可观测性较为完善，无需从服务端变量进行查询。如需定制监控工具，可阅读 [TiDB 监控框架概述](/tidb-monitoring-framework.md)来获得更多信息。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+
+### 不支持 UpdatableResultSet
+
+**描述**
+
+TiDB 暂不支持 UpdatableResultSet，即请勿指定 `ResultSet.CONCUR_UPDATABLE` 参数，并在 ResultSet 内部进行数据更新。
+
+**规避方法**
+
+使用额外的 `UPDATE` 语句进行数据更新，可使用事务保证数据一致性。
```
Comment (hfxsd) [inline develop/dev-guide-third-party-tools-compatibility.md:127]: 这里额外的是指什么？如果没有特别的含义可以删掉。

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 28]
```diff
@@ -0,0 +1,158 @@
...
+
+## MySQL JDBC Bug
+
+### `useLocalTransactionState` 和 `rewriteBatchedStatements` 同时开启将导致事务无法提交
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:131]:
```suggestion
### `useLocalTransactionState` 和 `rewriteBatchedStatements` 同时开启将导致事务无法提交或回滚
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 29]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+在 TiDB 应用程序中，请注意使用各语言的 64 位整型类型（或字符串类型）存储 `SELECT CONNECTION_ID()` 的结果，防止溢出。如 Java 应使用 `Long` 或 `String` 进行接收，JavaScript/TypeScript 应使用 `string` 类型进行接收。
+
+### TiDB 未设置 `Com_*` 计数器
+
+**描述**
+
+MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 `SHOW GLOBAL STATUS LIKE 'Com_%'` 观察 TiDB 与 MySQL 的差异。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:38]:
```suggestion
MySQL 维护了一系列 [`Com_` 开头的服务端变量](https://dev.mysql.com/doc/refman/8.0/en/server-status-variables.html#statvar_Com_xxx)来记录你对数据库的操作总数，如 `Com_select` 记录了 MySQL 数据库从上次启动开始，总共发起的 `SELECT` 语句数（即使语句并未成功执行）。而 TiDB 并未维护此变量。你可以使用语句 [`SHOW GLOBAL STATUS LIKE 'Com_%'`](/sql-statements/sql-statement-show-status.md) 观察 TiDB 与 MySQL 的差异。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 30]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+TiDB 不支持 `CHECK TABLE` 语句。
+
+**规避方法**
+
+在 TiDB 中使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。
+
+## MySQL JDBC 不兼容
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:64]:
```suggestion
## 与 MySQL JDBC 兼容性说明
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 31]
```diff
@@ -0,0 +1,158 @@
...
+
+**描述**
+
+TiDB 不支持 `CHECK TABLE` 语句。
+
+**规避方法**
+
+在 TiDB 中使用 [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) 语句进行表中数据和对应索引的一致性校验。
+
+## 与 MySQL JDBC 兼容性说明
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:64]:
```suggestion
## 与 MySQL JDBC 的兼容性
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 32]
```diff
@@ -0,0 +1,158 @@
...
+MySQL Connector/J 8.0.29 在与 5.7.5 版本以下的 MySQL 服务端，或使用 5.7.5 版本以下 MySQL 服务端协议的数据库（如 TiDB 6.3.0 版本以下）同时使用时，将在某些情况下导致数据库连接的挂起。关于更多细节信息，可查看此 [Bug Report](https://bugs.mysql.com/bug.php?id=106252)。
+
+**规避方法**
+
+这是一个已知的问题，截至 2022 年 10 月 12 日，MySQL Connector/J 未合并修复代码。
+
+TiDB 对其进行了两个维度的修复：
+
+- 客户端方面：可使用 [pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j) 替换官方的 MySQL Connector/J。**pingcap/mysql-connector-j** 中修复了该 Bug。
+- 服务端方面：可升级服务端至 6.3.0 版本或以上。TiDB 在 6.3.0 版本修复了此兼容问题。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:158]:
```suggestion
- 客户端方面：**pingcap/mysql-connector-j** 中修复了该 Bug，你可以使用 [pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j) 替换官方的 MySQL Connector/J。
- 服务端方面：TiDB v6.3.0 修复了此兼容性问题，你可以升级服务端至 v6.3.0 或以上版本。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 33]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+手动设置排序规则，不要依赖客户端默认排序规则。客户端默认排序规则由 MySQL Connector/J 配置文件保存。
+
+### 参数 `NO_BACKSLASH_ESCAPES` 不生效
+
+**描述**
+
+无法使用 `NO_BACKSLASH_ESCAPES` 参数从而不进行 `\` 字符的转义。已提 [issue](https://github.com/pingcap/tidb/issues/35302)。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:90]:
```suggestion
TiDB 中 无法使用 `NO_BACKSLASH_ESCAPES` 参数从而不进行 `\` 字符的转义。已提 [issue](https://github.com/pingcap/tidb/issues/35302)。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 34]
```diff
@@ -0,0 +1,158 @@
...
+
+**规避方法**
+
+不搭配使用 `NO_BACKSLASH_ESCAPES` 与 `\`，而是使用 `\\` 进行 SQL 编写。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:94]:
```suggestion
在 TiDB 中不搭配使用 `NO_BACKSLASH_ESCAPES` 与 `\`，而是使用 `\\` 编写 SQL 语句。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 35]
```diff
@@ -0,0 +1,158 @@
...
+
+| 字符集 | 客户端默认排序规则 | 服务端默认排序规则 |
+| --------- | -------------------- | ------------- |
+| `ascii`   | `ascii_general_ci`   | `ascii_bin`   |
+| `latin1`  | `latin1_swedish_ci`  | `latin1_bin`  |
+| `utf8mb4` | `utf8mb4_0900_ai_ci` | `utf8mb4_bin` |
+
+**规避方法**
+
+手动设置排序规则，不要依赖客户端默认排序规则。客户端默认排序规则由 MySQL Connector/J 配置文件保存。
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-tools-compatibility.md:84]:
```suggestion
在 TiDB 中手动设置排序规则，不要依赖客户端默认排序规则。客户端默认排序规则由 MySQL Connector/J 配置文件保存。
```

**Review by Oreoxmt — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]
LGTM

**Pull request merged**